core/src/main/scala/kafka/log/ProducerStateManager.scala

```diff
@@ -279,7 +265,7 @@ private[log] class ProducerAppendInfo(val topicPartition: TopicPartition,
              lastOffset: Long,
              isTransactional: Boolean): Unit = {
     val firstOffset = firstOffsetMetadata.messageOffset
-    maybeValidateAppend(epoch, firstSeq, firstOffset)
+    maybeValidateDataBatch(epoch, firstSeq, firstOffset)
```

Comment:
nit: maybe rename this `append` to `appendRecordBatch`?

Reply:
I went with `appendDataBatch`. Sound ok?

Reply:
Yea